### ISSUE TITLE
[Plugin] Show stat requirements of a ride in the coaster stat window in-game

### ISSUE THREAD

**Issue by FlyingThunder** [OpenRCT2/OpenRCT2#13520]
One of the biggest issue that complex games with a lot of background mechanics have is that in order to not make mistake, players have to rely on third party information for hidden mechanics. I am pretty sure that a lot of people dont even know that stat requirements exist (for example top speed, minimum length, amount of drops etc.) and are wondering why their rides have bad stats sometimes. So, i think a plugin that simply highlights the stats that are required and colors them green if met, and another color, maybe yellow (as red is used for excessive G forces) if not met, would be a huge improvement, and i can imagine that if the requirements are easily accessible in the code, it would be quite simple and uncomplicated.

![grafik](https://user-images.githubusercontent.com/39583733/101243095-2842ed80-36fe-11eb-8d17-e8b464b20cb3.png)

However, the stat for G forces are already colored in when exceeding certain limits, and some coasters have G forces as a stat requirement, so to not have complicated color coding, i can also imagine the slightly more complicated solution of adding a small icon next to the stat requirement, that could also show what stat is required when hovering over, like so (stats are just an example)
![grafik](https://user-images.githubusercontent.com/39583733/101243326-bc618480-36ff-11eb-9177-6fd03e12b56d.png)

**Comment by IntelOrca** [comment]
Plugins can't currently modify existing windows and the work to allow it is potentially quite a lot. It might just be easier to add this functionality to the base game, since it already highlights bad G forces in red.

**Comment by FlyingThunder** [comment]
@IntelOrca Good to know, i did not knew this, i thought that feature requests like these should only be made for plugins, but i can imagine this being in the base game aswell, thanks

**Comment by ocalhoun6** [comment]
Some coasters include requirements that aren't listed in the stats. Such as the reverser coaster requiring at least one reverser element.

**Comment by FlyingThunder** [comment]
good point, could this not be added as a new stat in a new line fairly easy? or do the stats have to be the same for all rides? as in, cant you add a line for reverser elements in the reverser coaster, or does that not work because the other rides dont have that element?

**Comment by ocalhoun6** [comment]
Not all rides have stats for air time, and the stat for inversions only shows up if the coaster has at least one inversion (if I remember right). Non-coaster rides (like a log flume) don't even have stats for G-forces at all. So it's definitely possible to have different stats visible for different rides. I'm not sure how much of that is hard-coded or embedded into the objects themselves, though. It might be difficult to change.

------------

On a related note... If you wanted to do something like this as a plugin, it would be *amazing* to have a plugin that gives an analysis of where the coaster's stats are coming from, going through the calculation step-by-step with a running total to the side. Like:

- length: 1902m | +4.32 excitement +1.67 intensity +0.00 nausea | 4.32e, 1.67i, 0.00n
- max speed: 64kph | +3.25 excitement +3.40 intensity +1.50 nausea | 7.57e, 5.07i, 1.50n
- number of drops 27 | +5.00 excitement +8.43 intensity +6.79 nausea | 12.57e, 13.50i, 8.29n
- small helix element | +0.15 excitement +0.00 intensity +0.20 nausea | 12.72e, 13.50i, 8.49n
- small helix element | +0.15 excitement +0.00 intensity +0.20 nausea | 12.87e, 13.50i, 8.69n
- small helix element | +0.15 excitement +0.00 intensity +0.20 nausea | 13.02e, 13.50i, 8.89n
- track elements between 1 and 2 units above water bonus | +0.30 excitement, +0.00 intensity +0.20 nausea | 13.32e, 13.50i, 9.09n
- excessive intensity penalty | excitement divided by 2 | 6.66e, 13.50i, 9.09n
- missed stat requirement penalty (no inversions) | all stats divided by 2 | 3.33e, 6.75i, 4.55n

(Obviously, this isn't accurate to the game's actual calculations, just an illustration to get the point across.)
That would be a pretty awesome plugin.

**Comment by josepharhar** [comment]
I’d also like a way to know if I’m missing a stat requirement. Building coasters without actually knowing sounds frustrating and is the reason I just download templates for rides.

**Comment by ThiaJay** [comment]
This sounds awesome.
